### DISCUSSION TITLE
调用方法的时候错误建议是用reject还是throw抛出？

### DISCUSSION THREAD

**Discussion by xmsz** [midwayjs/midway#1562] [Q&A]
背景
- 比如service的错误
- 是用Promise.reject抛出，还是直接用throw(MidwayError)比较好

还有
- 如果使用throw是直接throwHttp错误还是普通错误，让外层再拦截

或者
- 什么情况下推荐用Promise.reject抛出错误，什么时候用throw

还有
- 我throw以后，请求返回的内容并没有带上错误信息，我还需要配置什么来让返回的内容有具体的错误原因。我现在是https://www.midwayjs.org/docs/next/error_filter#%E5%BC%82%E5%B8%B8%E5%A4%84%E7%90%86%E5%99%A8 自定义了一个

**Comment by waitingsong** [comment]
- 通常情况用 `throw` 
- 在 setTimeout 之类脱离上下文的代码内应该考虑用外层的 Promise.reject()

**Reply by xmsz** [reply]
ok
